### PR TITLE
[Fix] Support TP for ModelOpt NVFP4 by adding dynamic padding

### PR DESCRIPTION
**Pull request by danielafrimi**
## Summary
This PR adds dynamic padding support for NVFP4-quantized linear layers to handle tensor parallelism (TP) cases where partitioned weights don't meet kernel alignment requirements.

## Problem
When using TP > 2 with NVFP4 quantization, weight partitions may not satisfy the alignment constraints required by FlashInfer and Cutlass kernels (e.g., 32-row tile alignment, group-size block alignment). This caused failures or incorrect results.

### CODE REVIEW

File: vllm/model_executor/layers/quantization/modelopt.py [hunk 1]
```diff
@@ -1055,9 +1056,91 @@ def process_weights_after_loading(self, layer: Module) -> None:
...
+            # Swizzle block scales and then pad the packed NVFP4 weights so that
+            # both N (rows) and K (columns) satisfy the alignment constraints
+            # implied by the FlashInfer / Cutlass kernels.
+            swizzled_weight_scale = swizzle_blockscale(layer.weight_scale)  # [2576, 168] -> [2688, 168]
             layer.weight_scale = Parameter(swizzled_weight_scale, requires_grad=False)
-            layer.weight = Parameter(layer.weight.data, requires_grad=False)
+            weight = layer.weight.data
+
+            # Zero-pad weights in N dim if swizzled block scales have more rows
+            # than the original weights (e.g. to satisfy 32-row tile alignment).
```
Comment (pavanimajety) [inline vllm/model_executor/layers/quantization/modelopt.py:1067]: Could you please share the error log where the kernels require this?

**Comment by hmellor** [comment]
Closing as superseded by #31099, which landed the same NVFP4 TP padding (`pad_nvfp4_weight_for_cutlass`, `pad_nvfp4_activation_for_cutlass`, `slice_nvfp4_output`). The code this draft patches has also since been refactored into `vllm/model_executor/kernels/linear/nvfp4/`.
